exchangelib/version.py

```diff
@@ -225,14 +232,14 @@ def _guess_version_from_service(cls, protocol, hint=None):
     @classmethod
     def _get_version_from_service(cls, protocol, api_version):
         assert api_version
+        header = None
         xml = dummy_xml(version=api_version)
         # Create a minimal, valid EWS request to force Exchange into accepting the request and returning EWS xml
         # containing server version info. Some servers will only reply with their version if a valid POST is sent.
         session = protocol.get_session()
         log.debug('Test if service API version is %s using auth %s', api_version, session.auth.__class__.__name__)
         r, session = post_ratelimited(protocol=protocol, session=session, url=protocol.service_endpoint, headers=None,
-                                      data=xml, timeout=protocol.TIMEOUT, verify=protocol.verify_ssl,
-                                      allow_redirects=False)
+                                      data=xml, timeout=protocol.TIMEOUT, verify=protocol.verify_ssl)
```

Comment:
`allow_redirects=False` is intentional - we want full control over redirection here. Please add it back.